ktor-server/ktor-server-plugins/ktor-server-metrics-micrometer/jvm/test/io/ktor/server/metrics/micrometer/MicrometerMetricsTests.kt

```diff
@@ -267,29 +253,31 @@ class MicrometerMetricsTests {
             assertTrue(noHandlerHandledRequest)
         }
 
-    private fun TestApplicationEngine.installDefaultBehaviour() {
-        this.callInterceptor = {
-            try {
-                call.application.execute(call)
-                if (call.response.status() == HttpStatusCode.NotFound) {
-                    noHandlerHandledRequest = true
-                }
-            } catch (t: Throwable) {
-                throwableCaughtInEngine = t
-                if (call.response.status() == null) {
-                    call.respond(HttpStatusCode.InternalServerError)
+    private fun ApplicationTestBuilder.installDefaultBehaviour() {
+        application {
+            (engine as TestApplicationEngine).callInterceptor = {
```

Comment:
Is there a better way to specify `callInterceptor` without casting?

Reply:
You can use the fuinction `intercept`:
```kotlin
        application {
            val beforeSetup = PipelinePhase("BeforeSetup")
            insertPhaseBefore(ApplicationCallPipeline.Setup, beforeSetup)
            intercept(beforeSetup) {
                try {
                    proceed()
                    if (call.response.status() == HttpStatusCode.NotFound) {
                        noHandlerHandledRequest = true
                    }
                } catch (t: Throwable) {
                    throwableCaughtInEngine = t
                    if (call.response.status() == null) {
                        call.respond(HttpStatusCode.InternalServerError)
                    }
                }
            }
        }
```
It's important to add phase `beforeSetup` because `catch` here should work after `catch` in hook `CallFailed` (which is used in `MicrometerMetrics`), and `CallFailed` is set before `Setup` phase